### PR TITLE
test: Revert sys-fs-bpf.mount rename

### PR DESCRIPTION
**Pull request by jrajahalme**
Renaming `sys-fs-bpf.mount` to `cilium-sys-fs-bpf.mount` made the mount
inoperable, but this surfaced only in backports.

Revert the rename, release branches already have this fix as part of the
backports.

Fixes: #19310.
Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

### CODE REVIEW

File: test/provision/compile.sh [hunk 1]
```diff
@@ -136,7 +136,7 @@ else
     # Do not run cilium-operator with runtime tests, as it fails to connect to k8s api-server
     rm -f /etc/systemd/system/cilium-operator.service
 
-    services=$(cd /etc/systemd/system; ls -1 cilium*.service cilium*.mount)
+    services=$(cd /etc/systemd/system; ls -1 cilium*.service sys-fs-bpf.mount)
```
Comment (pchaigno) [inline test/provision/compile.sh:139]: Why was this file originally renamed and what's the issue that requires this revert?

Comment (jrajahalme) [reply test/provision/compile.sh:139]: @pchaigno Prior to the change to use docker for running Cilium in CI runtime tests all service files (`*.*`) in `contrib/systemd/` were listed to form a list of services and mounts that need to be enabled and started. As part of this change I needed to add an alternative `cilium.service` (`cilium.service-with-docker`) and eliminate `cilium-operator.service` from the runtime test setup, but I did not want these changes to have any effect on uses other than CI runtime tests. As such I could not delete the old files still needed to run cilium via systemd as before, and it was no longer possible to form the list of services to be enabled and started by listing files in `contrib/systemd`. Instead, the needed files are listed from `/etc/systemd/system/` after they are copied over from `contrib/systemd/` to `/etc/systemd/system/` (with `cilium.service-with-docker` copied over as `cilium.service` and `cilium-operator.service` deleted). I needed a simple way to only list Cilium files from there and thought that all of them having a `cilium`-prefix would do the trick.

The problem with this is that `sys-fs-bpf.mount` does not work when renamed as `cilium-sys-fs-bpf.mount`. Apparently systemd mounts encode the path in the name (`sys-fs-bpf.mount` corresponds to path `/sys/fs/bpf`, and `cilium-sys-fs-bpf.mount` would correspond to path `/cilium/sys/fs/bpf`), hence the renamed mount did not work:

> Mount units must be named after the mount point directories they control. Example: the mount point /home/lennart must be configured in a unit file home-lennart.mount. For details about the escaping logic used to convert a file system path to a unit name, see [systemd.unit(5)](https://www.freedesktop.org/software/systemd/man/systemd.unit.html#). Note that mount units cannot be templated, nor is possible to add multiple names to a mount unit by creating additional symlinks to it.

Master branch CI does not need this mount file so this was noticed only in the backports.

This revert is already done in the backports, so this PR is the corresponding fix for the master branch. The issue why this revert is needed is that e.g., RPM packaging of Cilium (see `contrib/packaging/rpm`) uses the mount file and likely expects the mount to be correctly named.

Comment (pchaigno) [reply test/provision/compile.sh:139]: Thanks for the detailed explanation! :rocket: Would it be possible to add it to the commit description for future reference?

**Review by nbusseneau — APPROVED** [review]
Code changes LGTM.

File: test/provision/compile.sh (hunk 1 above)
Comment (jrajahalme) [reply test/provision/compile.sh:139]: Added detailed reasoning to the commit message as requested by @pchaigno

**Comment by jrajahalme** [comment]
Force pushed to update the commit message. This change does not affect the CI, so there is no point re-doing the tests. Marking as ready-to-merge and release-blocker-1.12 as any possible downstream rpm packaging is broken without this revert.

**Review by pchaigno — APPROVED** [review]

**Pull request merged**